### PR TITLE
Implement version 3 spec of the test runner

### LINKED ISSUE #32: Upgrade to version 3 spec
If possible, this test runner should be updated to version 3 of the [test runner interface specification](https://github.com/exercism/docs/blob/main/building/tooling/test-runners/interface.md). In version 3, one additional feature is enabled: the ability to link individual tests to tasks. This allows the website to show which tests belong to which tasks.

The way tests are linked to tasks is via an (optional) `task_id` field, which is an integer that matches the number of the task as defined in the exercise's `instructions.md` file (note: the instructions start at index 1).

This is an example of a test in the `results.json` file:

```json
{
  "name": "Expected oven time in minutes",
  "status": "pass",
  "task_id": 1,
  "test_code": "Assert.Equal(40, Lasagna.ExpectedMinutesInOven());"
}
```

You are completely free in how to implement this. Some options are:

1. Add metadata to a test that the test runner can then discover while running the tests (e.g. an attribute or annotation)
2. Define a test name/task id mapping (e.g. in the exercise's `.meta/config.json` file)
3. Any other option you can think of...

Let me know if there are any questions.

### PR DESCRIPTION
**Pull request by junedev**
Resolves #32

### CODE REVIEW

**Comment by junedev** [comment]
@ErikSchierboom Is there any other place then the JSON the test runner outputs where I need to specifiy that it is version 3 now?

**Comment by ErikSchierboom** [comment]
@junedev Nope, we've only dabbled with other locations for the version with representers.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by junedev** [comment]
@ErikSchierboom I updated the code as discussed so either all or none of the tests have task ids. Do you want to take another look?

File: README.md [hunk 1]
```diff
@@ -170,4 +170,60 @@ This is done via the `.meta/config.json` file of the exercise. See example below
...
+```
+
+Note that this flag should only ever be set for concept exercises.
+Task ids are currently not supported for practice exercise.
+
+There are two ways the test runner can assign task ids for concept exercises.
+
+### Implicit Task Id Assignment
+
+If `"enableTaskIds": true` was set but there are no explicit task ids found in the test file, the test runner will automatically assign task ids.
```
Comment (ErikSchierboom) [inline README.md:199]: Should this be:

```suggestion
If `"taskIdsEnabled": true` was set but there are no explicit task ids found in the test file, the test runner will automatically assign task ids.
```

**Review by ErikSchierboom — APPROVED** [review]
It's slightly hard to review due to not seeing the generated `results.json`, but I'm happy to approve

**Comment by junedev** [comment]
> It's slightly hard to review due to not seeing the generated results.json, but I'm happy to approve

@ErikSchierboom Yes, I would also prefer these "integration tests" but that's for another time.

**Pull request merged**